sdk/resources/Azure.ResourceManager.Resources/src/Generated/Extensions/MockingExtensions.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq.Expressions;
+using System;
+using Moq;
+using Mock = Moq.Mock;
+using System.Reflection;
+
+namespace Azure.ResourceManager.Resources.Mocking
+{
+    internal static class MockingExtensions
+    {
+        internal static void SetupAzureExtensionMethod<T>(this Mock<T> mock, Expression<Action<T>> expression) where T : ArmResource
```

Comment:
What if a customer is using a different mocking library?   For example, `NSubstitute`, `FakeItEasy`, `RhinoMocks`, `JustMock`

We do not want to offer built-in support for a specific mocking library and remove customer choice or provide a degraded experience for those who prefer other libraries.